### DISCUSSION TITLE
Secondary Virtual COM Port on Raspberry Pi Pico

### DISCUSSION THREAD

**Discussion by omicronns** [micropython/micropython#9790] [Using MicroPython]
Hi,

I'm trying to add secondary VCP in MicroPython on Raspberry Pi Pico. I found [this](https://forum.micropython.org/viewtopic.php?t=6594) thread, and followed it. Unfortunately it doesn't work. I can build micropython with additional entries in `micropython/ports/rp2/boards/PICO/mpconfigboard.h`:

```
#define MICROPY_HW_USB_CDC_NUM      (2)
#define MICROPY_HW_USB_ENABLE_CDC2  (1)
```

But no additional com port is present in my system, only `ttyACM0`. Next point in linked thread is about adding `boot.py`, but `pyb` object/module is not available for me.

What am I doing wrong?

**Comment by jimmo** [comment]
As described in that forum post, this feature is only for the stm32/pyboard (and only the high-end ones).

We're working on more support for configuring the USB endpoints for Pico and other ports (including HID support), but this is not ready yet

**Reply by omicronns** [reply]
Thank you for the response. It will be really useful feature. Looking forward to it.

**Comment by barafael** [comment]
If I may ask, is there anything going on with this subject? I am also interested in opening a second VCP.

**Reply by scruss** [reply]
this is the PR to watch for progress: [tinyusb: Support USB device drivers written in Python by projectgus · Pull Request #9497 · micropython/micropython](https://github.com/micropython/micropython/pull/9497)
